contracts/facets/ArchaeologistFacet.sol

```diff
@@ -34,22 +48,88 @@ contract ArchaeologistFacet {
         uint256 withdrawnReward
     );
 
+    function registerArchaeologist(
+        uint256 minimumDiggingFee,
+        uint256 maximumRewrapInterval,
+        uint256 freeBond
+    ) external {
+        // verify that the archaeologist does not already exist
+        LibUtils.revertIfArchProfileExists(msg.sender);
+
+        // create a new archaeologist
+        LibTypes.ArchaeologistProfile memory newArch =
+            LibTypes.ArchaeologistProfile({
+                exists: true,
+                minimumDiggingFee: minimumDiggingFee,
+                maximumRewrapInterval: maximumRewrapInterval,
+                freeBond: freeBond,
+                cursedBond: 0,
+                rewards: 0
+            });
+
+        // transfer SARCO tokens from the archaeologist to this contract, to be
+        // used as their free bond. can be 0.
+        if (freeBond > 0) {
+            s.sarcoToken.transferFrom(msg.sender, address(this), freeBond);
```

Comment:
Is there a way we could easily integrate approval into this process? Anytime this is called without previously having approved spending, this will revert - which I think will happen nearly 100% of the time.

Actually as I'm typing this I'm thinking this is something that the arch service will have to remember to do prior to attempting to register with free bond.

Reply:
@DarksightKellar -- yeah, that's a great point. When registering/updating + `freeBond` > 0, the arch service should first check to see if their wallet has an approval set on the SARCO token contract. If it doesn't, it should create an approval TX (and wait for it to confirm) prior to calling register/update.